pkg/engine/task/task_apply.go

```diff
@@ -93,6 +99,16 @@ func apply(ro []runtime.Object, c client.Client) ([]runtime.Object, error) {
 	return applied, nil
 }
 
+func isClusterResource(r runtime.Object) bool {
```

Comment:
hmm I think we need much better detection of this, there is much more cluster-scoped resources - ClusterRole, PersistentVolumes and many more probably

Reply:
+1 to that. This is already a fix, but I'd like to research whether we should reinvent the wheel here,

Reply:
yeah... I did some research but more is welcomed.   this does fix the issue tho and we should get it to an acceptable state and merge it.   Can we evolve this based on research after ?